### PR TITLE
Refining example, add utilities, and fix xdist test error

### PR DESCRIPTION
**Pull request by loomlike**
## Description
This PR contains - utility functions, fix bugs, and refactor quick start (nyc taxi) notebooks.
Resolves #792 , revert #798

## How was this PR tested?
Unit tests including the sample notebook test.
In a local machine, a test w/ the xdist option passed.

## Does this PR introduce any user-facing changes?
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

**Review by blrchen — APPROVED** [review]

File: feathr_project/feathr/client.py [hunk 1]
```diff
@@ -415,7 +415,6 @@ def get_offline_features(self,
                              output_path: Union[str, Sink],
                              execution_configurations: Union[SparkExecutionConfiguration ,Dict[str,str]] = {},
                              config_file_name:str = "feature_join_conf/feature_join.conf",
-                             udf_files = None,
```
Comment (Yuqing-cat) [inline feathr_project/feathr/client.py:418]: Any specific reason to remove this param from `client.py`?

**Review by Yuqing-cat — COMMENTED** [review]
@blrchen  would you help to confirm if the test result is as expected?
`
databricks:====== 12 failed, 72 passed, 7 skipped, 74 warnings in 873.66s (0:14:33) =======`
`synapse:====== 22 failed, 63 passed, 6 skipped, 16 warnings in 1478.44s (0:24:38) ======`

The latest readable CI run doesn't have that much failures: https://github.com/feathr-ai/feathr/actions/runs/3279090567/jobs/5398225684

**Review by Yuqing-cat — APPROVED** [review]
@blrchen , let's merge it first to unblock CI testing issue. But remember to double confirm about the `udf_file` parameters of `client.py` later.

**Comment by blrchen** [comment]
> `databricks:====== 12 failed, 72 passed, 7 skipped, 74 warnings in 873.66s (0:14:33) =======`
> `synapse:====== 22 failed, 63 passed, 6 skipped, 16 warnings in 1478.44s (0:24:38) ======`

This PR only resolves the issue that test not work with pytest-xdist. For the 12 databricks failures and 22 synapse, they are related to #756. The failed test case # before this PR is before about 5 or 7)

@loomlike @Yuqing-cat i think we should revert #756 from main now for 0.9.0 release. @loomlike can submit it again to next release after all CI test failures are resolved.

**Comment by loomlike** [comment]
Sorry for the large file changes. It's due to reverting #798.
Additionally, this PR includes changes in #786

File: feathr_project/feathr/client.py (hunk 1 above)
Comment (loomlike) [reply feathr_project/feathr/client.py:418]: The variable `udf_files` is assigned in the body of the function which will override the args anyway:
```
        udf_files = _PreprocessingPyudfManager.prepare_pyspark_udf_files(feature_names, self.local_workspace_dir)
```

**Review by xiaoyongzhu — DISMISSED** [review]
This looks good to me. Please help fix the conflicts.

**Review by blrchen — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Review by blrchen — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**